src/core/auth/options.ts

```diff
@@ -56,6 +56,8 @@ export interface AuthOptionsT<AuthHandler> {
   endpoint: string;
   params?: any;
   headers?: any;
+  paramProvider?: () => any;
+  headerProvider?: () => any;
```

Comment:
Regarding the naming, I don't have a better alternative at the moment but here are the changes I suggest for this name:
- `paramProvider` => `paramsProvider`
- `headerProvider` => `headersProvider`

To indicate that it returns multiple params and multiple headers.